### DISCUSSION TITLE
Как импортировать данные из excel?

### DISCUSSION THREAD

**Discussion by ivgo** [totumonline/totum-ru-issues-and-discussions#332] [Как это сделать?] [answered]
Я нашел для себя один из самых простых способов импорта данных.
Для этого нам понадобиться n8n

**Comment by ivgo** [comment, marked as answer]
Создал простую таблицу с 3 полями
![grafik](https://user-images.githubusercontent.com/10232491/224510363-e1cca3cb-7df7-4ab0-b917-cf22a2d43076.png)

Код Кнопки:

```
= :getFromScript(uri: "https://*********"; post:   "file" = #h_linkto; )
```

Ячейка с файлом:

Код Поля со ссылкой:

```
=: str`"https://" + $#nh + "/fls/" + #h_files[0][file]`
```

Далее нам понадобится n8n делаем примерно так.

![grafik](https://user-images.githubusercontent.com/10232491/224510768-9a05bc38-3cb3-4f56-8d57-774c2fd4fa53.png)


1. webhook которому мы передаем ссылку для скачивания файла
2. Создаем HTTP Request для скачивания файла
3. Далее SpreadsheetFile читаем скаченное 
4. Через трасформацию данных собираем наш массив 
5. Создаем HTTP Request и передаем наш JSON

А далле из урока про API присваиваю значение полю Данные
получаем что то вроде :
```
{
  "Feld_a": "a",
  "Feld_b": "b",
  "Feld_c": "c",
  "Feld_d": "d",
  "Feld_e": "e"
}
```

Разбираем и вставляем куда надо

второй способ Ява оствлю ссылку там просто : 

https://www.youtube.com/watch?v=IcxJUJXRPGw

**Reply by ttm-alexeypolunin** [reply]
Лайк!
